### ISSUE TITLE
Dataset not found even though I thought I have written my paths in my data.yaml

### ISSUE THREAD

**Issue by roberthobblebottom** [ultralytics/yolov5#4561]
## ❔Question
```
WARNING: Dataset not found, nonexistent paths: ['/home/ryan/Documents/internship/yolov5-master/images/val']
Traceback (most recent call last):
  File "train.py", line 658, in <module>
    main(opt)
  File "train.py", line 559, in main
    train(opt.hyp, opt, device)
  File "train.py", line 129, in train
    check_dataset(data_dict)  # check
  File "/home/ryan/Documents/internship/yolov5-master/utils/general.py", line 243, in check_dataset
    raise Exception('Dataset not found.')
Exception: Dataset not found.
```

**Here is my data.yaml:**
```
path: ../dataset 
train: images/train
val: images/val
test:  # test images (optional)

# Classes
nc: 1  # number of classes
names: ["patient not in bed"]  # class e
```

My images are in `../dataset/images/train` and ''../dataset/images/val' respectively
and my labels are in ` ../dataset/labels/train` and ''../dataset/labels/val' respectively

What am I missing out here? Thanks for reading.

## Additional context
I went through [Train Custom Data](https://github.com/ultralytics/yolov5/wiki/Train-Custom-Data) page many times already, things to me seems correct. Did I skipped over something on accident?

Also tried just putting the images and labels in the same folder as yolov5-master but that gives out a different error about not able to find the labels even though the labels are in the other folder as mentioned.

**Comment by glenn-jocher** [comment]
@themissingCRAM follow Train Custom Data tutorial and use COCO128 as an example:
```
python train.py --data coco128.yaml
```

## YOLOv5 Tutorials

* [Train Custom Data](https://docs.ultralytics.com/yolov5/tutorials/train_custom_data)&nbsp; 🚀 RECOMMENDED
* [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/)&nbsp; ☘️ RECOMMENDED
* [Weights & Biases Logging](https://github.com/ultralytics/yolov5/issues/1289)&nbsp; 🌟 NEW
* [Supervisely Ecosystem](https://docs.ultralytics.com/yolov5/tutorials/model_export8)&nbsp; 🌟 NEW
* [Multi-GPU Training](https://docs.ultralytics.com/yolov5/tutorials/multi_gpu_training)
* [PyTorch Hub](https://docs.ultralytics.com/yolov5/tutorials/pytorch_hub_model_loading)&nbsp; ⭐ NEW
* [TorchScript, ONNX, CoreML Export](https://docs.ultralytics.com/yolov5/tutorials/model_export) 🚀
* [Test-Time Augmentation (TTA)](https://docs.ultralytics.com/yolov5/tutorials/test_time_augmentation)
* [Model Ensembling](https://docs.ultralytics.com/yolov5/tutorials/model_ensembling)
* [Model Pruning/Sparsity](https://docs.ultralytics.com/yolov5/tutorials/model_pruning_and_sparsity)
* [Hyperparameter Evolution](https://docs.ultralytics.com/yolov5/tutorials/hyperparameter_evolution)
* [Transfer Learning with Frozen Layers](https://docs.ultralytics.com/yolov5/tutorials/transfer_learning_with_frozen_layers)&nbsp; ⭐ NEW
* [TensorRT Deployment](https://github.com/wang-xinyu/tensorrtx)

**Comment by themissingCRAM** [comment]
That means I put my custom data on top of the COCO128 dataset?

**Comment by roberthobblebottom** [comment]
I think this particular problem is solved when I tried using my custom dataset with coco128. But I am facing another problem trying to train right now. It suddenly killed the training half way: 
![image](https://user-images.githubusercontent.com/60025340/131246872-ac5af1c6-c256-4e49-aa67-15c88c0740f3.png)

![image](https://user-images.githubusercontent.com/60025340/131243343-e21c910d-2b26-49a5-ae00-010a82dd06a0.png)

Should I close this issue and open another one?

Or maybe I have too much missing labels and should remove some?

How do I make sure that the train.py knows that those are background images? I know I know I need to remove some to reduce it to 10% of the dataset.


Edit: I just tried removing all background images but train.py killed itself again. According to another post is this because of my environment somehow? I already reclone the repo yet it gave me this error and `$ pip install -r requirements.txt` says that my requirements are already satisfied.

**Comment by themissingCRAM** [comment]
![image](https://user-images.githubusercontent.com/60025340/131243497-69c348b3-b897-41df-8fbc-fb02791e6199.png)


Here is the rest of the log if anyone is interested to help me in this issue post

**Comment by glenn-jocher** [comment]
@themissingCRAM it appears you may have environment problems. Please ensure you meet all dependency requirements if you are attempting to run YOLOv5 locally.  If in doubt, create a new virtual Python 3.8 environment, clone the latest repo (**code changes daily**), and `pip install -r requirements.txt` again. We also highly recommend using one of our verified environments below. 


## Requirements

[**Python>=3.6.0**](https://www.python.org/) with all [requirements.txt](https://github.com/ultralytics/yolov5/blob/master/requirements.txt) installed including [**PyTorch>=1.7**](https://pytorch.org/get-started/locally/). To get started:
```bash
$ git clone https://github.com/ultralytics/yolov5
$ cd yolov5
$ pip install -r requirements.txt
```

## Environments

YOLOv5 may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Google Colab and Kaggle** notebooks with free GPU: <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/yolov5"><img src="https://img.shields.io/docker/pulls/ultralytics/yolov5?logo=docker" alt="Docker Pulls"></a>


## Status

![CI CPU testing](https://github.com/ultralytics/yolov5/workflows/CI%20CPU%20testing/badge.svg)

If this badge is green, all [YOLOv5 GitHub Actions](https://github.com/ultralytics/yolov5/actions) Continuous Integration (CI) tests are currently passing. CI tests verify correct operation of YOLOv5 training ([train.py](https://github.com/ultralytics/yolov5/blob/master/train.py)), validation ([val.py](https://github.com/ultralytics/yolov5/blob/master/val.py)), inference ([detect.py](https://github.com/ultralytics/yolov5/blob/master/detect.py)) and export ([export.py](https://github.com/ultralytics/yolov5/blob/master/export.py)) on MacOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by roberthobblebottom** [comment]
I  believe I have the docker container for yolov5 and tried it out. Unfortunately as of now it is killing the process. I will need to check the guides again and I don't want to use other environments just in case if they are peaking into my data.


I will update on this comment.

Running on a virtual machine that I have, I found out that I have ran out of memory. So going back to host computer, I tried train.py and the memory did shot up and some points, reached 99% of memory usage. It may be obvious for some but that is the point where my venture went wrong and it is good to learn from it.
